### PR TITLE
internal/v5: return more info from new upload

### PR DESCRIPTION
**Pull request by rogpeppe**
This populates the NewUploadResponse with information
that the multipart client requires.

### CODE REVIEW

File: internal/v5/upload_test.go [hunk 1]
```diff
@@ -37,19 +38,36 @@ func (s *APISuite) TestPostUpload(c *gc.C) {
...
-	if want := now.Add(2 * time.Minute).Truncate(time.Millisecond); info.Expires.Before(want) {
-		c.Errorf("expires too early, got %v, want %v", info.Expires, want)
-	}
-	if want := now.Add(2*time.Minute + 5*time.Second); info.Expires.After(want) {
-		c.Errorf("expires too late, got %v, want %v", info.Expires, want)
-	}
+	c.Assert(info.Expires.UTC(), gc.Equals, expires.UTC())
+}
+
+func assertTimeBetween(c *gc.C, t, t0, t1 time.Time) {
```
Comment (fabricematrat) [inline internal/v5/upload_test.go]: remove this probably.

**Review by fabricematrat — APPROVED** [review]

**Review by mhilton — APPROVED** [review]
LGTM

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2055/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2056/

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Pull request merged**